ogr/ogr_srs_esri.cpp

```diff
@@ -589,8 +627,18 @@ OGRErr OGRSpatialReference::importFromESRI(char **papszPrj)
                 }
                 if (!bFoundParameters)
                 {
-                    // If unknown, default to WGS84 so there is something there.
-                    SetWellKnownGeogCS("WGS84");
+                    if (bHasRadiusOfSphereOfReference)
+                    {
+                        OGRSpatialReference oGCS;
+                        oGCS.SetGeogCS("unknown", "unknown", "unknown",
+                                       dfRadiusOfSphereOfReference, 0);
```

Comment:
What if the units in the file are not meters, is this radius of the earth in meters? I don't know.

For instance, I have seen files from Trimble where the radius of the ellipsoid is in the linear units of the file, that can be foot

Reply:
> What if the units in the file are not meters, is this radius of the earth in meters? I don't know.

importFromESRI() (seems to) assumes that Units is only for the projected linear units, not for the ellipsoid dimension. I'm not aware of a specification for ArcInfo 7.x .prj file, so we can't know for sure until someone exhibits a counter example

> For instance, I have seen files from Trimble where the radius of the ellipsoid is in the linear units of the file

ArcInfo 7.x .prj files ?

Reply:
> > For instance, I have seen files from Trimble where the radius of the ellipsoid is in the linear units of the file
> 
> ArcInfo 7.x .prj files ?

No. Trimble Site Calibration files (.CAL or .DC)